File: src/org/onion_lang/onion/compiler/env/StaticImportItem.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package org.onion_lang.onion.compiler.env;

/**
 * @author Kota Mizushima
 * Date: 2005/04/15
 */
public class StaticImportItem {
  private String name;
  private boolean fqcn;

  public StaticImportItem(String name, boolean fqcn) {
    this.name = name;
    this.fqcn = fqcn;
  }
  
  /**
   * returns name.
   */
  public String getName() {
    return name;
  }
  
  /**
   * returns whether getName() is FQCN or not.
   * @return
   */
  public boolean isFqcn() {
    return fqcn;
  }
  
  /**
   * matches getName() with name.
   * @param name
   * @return if name is matched, then return true.
   */
  public boolean match(String name) {
    if(fqcn){
      return this.name.equals(name);
    }else{
      return this.name.lastIndexOf(name) == this.name.length() - name.length();
    }
  }
}

File: src/main/scala/onion/compiler/StaticImportItem.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2016-, Kota Mizushima, All rights reserved.  *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler

/**
 * @author Kota Mizushima
 *
 */
class StaticImportItem(
  val name: String,
  val fqcn: Boolean,
  val methodName: String = null
) {

  /**
   * returns name.
   */
  def getName: String = name

  /**
   * returns whether name() is FQCN or not.
   * @return
   */
  def isFqcn: Boolean = fqcn

  /**
   * Returns the imported method name, or null if the whole class is imported.
   */
  def getMethodName: String = methodName

  /**
   * matches name() with name.
   * @param name
   * @return if name is matched, then return true.
   */
  def `match`(name: String): Boolean = {
    if (fqcn) {
      this.name.equals(name)
    }
    else {
      this.name.lastIndexOf(name) == this.name.length - name.length
    }
  }

  /**
   * Returns true if this import makes the given static method available.
   */
  def importsMethod(methodName: String): Boolean = {
    if (this.methodName == null) true
    else this.methodName == methodName
  }
}